### ISSUE TITLE
[CI Failure]: mi325_1: Samplers Test

### ISSUE THREAD

**Issue by AndreasKaratzas** [vllm-project/vllm#29521]
### Name of failing test

`pytest -v -s samplers && VLLM_USE_FLASHINFER_SAMPLER=1 pytest -v -s samplers`

### Basic information

- [ ] Flaky test
- [ ] Can reproduce locally
- [ ] Caused by external libraries (e.g. bug in `transformers`)

### 🧪 Describe the failing test

**Failing Tests Summary:**

**test_beam_search_with_concurrency_limit** in `test_beam_search.py`
- Tests: Beam search output consistency with/without concurrency limits
- Failure: Output mismatch between concurrent and sequential beam search
- Configuration: TinyLlama-1.1B, beam_width=4, max_tokens=64, dtype=half
- Likely cause: Non-deterministic behavior or race conditions in beam search implementation when processing requests concurrently, possibly related to scheduling or state management differences on ROCm.

**test_ranks** in `test_logprobs.py` (4 failures)
- Tests: Logprobs rank validation for greedy/non-greedy with flat/nested logprobs
- Failure: Assertion errors in logprobs structure or rank verification
- Configuration: distilgpt2, all combinations of greedy/flat_logprobs flags
- Likely cause: Systematic issue with logprobs calculation, extraction, or ranking logic, potentially related to numerical precision in fp16 on ROCm or incorrect logprobs tensor operations.

**test_two_token_bad_word** in `test_no_bad_words.py`
- Tests: Bad words filtering with multi-token sequences
- Failure: AssertionError - generated `[0, 0]` instead of expected `[812, 1468]`
- Configuration: distilgpt2, dtype=half, prompt="How old are you? I am 10"
- Likely cause: Token generation completely broken, producing padding/invalid tokens (0). Bad words filtering may be corrupting the sampling logits or the model isn't generating properly at all, suggesting fundamental issues with the sampling pipeline or logits processing.

### 📝 History of failing test

AMD-CI build Buildkite references: 
- 1041
- 1077
- 1088
- 1109
- 1111

**Comment by AndreasKaratzas** [comment]
@qli88 analyzing, vLLM can't run model "distilbert/distilgpt2" on graph mode.

**Comment by AndreasKaratzas** [comment]
PR is up for this group: https://github.com/vllm-project/vllm/pull/31873
